### ISSUE TITLE
Keyboard shortcuts

### ISSUE THREAD

**Issue by danoe** [kapitainsky/RcloneBrowser#94]
I'm running into this quite a lot, downloading single items or folders one by one. So I was thinking it would be great to have some keyboard shortcuts to help.

They should have a prefix, maybe even configurable, like CTRL, because currently we can navigate large folders with by typing the first letters of folders/files

From the browse windows something like CTRL+D or CTRL-U to open the Download / Upload window and then something like R or CTRL+R if prefix is global to run the job.

As this is basically the only workflow I use, I haven't thought more about it. Maybe someone else has more examples for shortcuts for different workflows.

**Comment by kapitainsky** [comment]
Definitely will add - should be easy.
Have you thought about using filtering?

--files-from or --include could make transferring multiple files easy.

**Comment by kapitainsky** [comment]
I would create task with --files-from mylist.txt then just go over files I have to transfer - copy/paste names. save file. and run task

**Comment by kapitainsky** [comment]
What OS you are on? I have added it already - will send you link to beta version with this feature included.

**Comment by kapitainsky** [comment]
I also looked into the code Alt-U, Alt-D and Alt-R should work on Windows.. it is another hell:) different OS have different keys

**Comment by danoe** [comment]
I can test on Windows and Linux :)

Filtering doesn't really work for me, i'm picking movie names, or a some single series episodes.

**Comment by kapitainsky** [comment]
Here you are latest Windows beta:

https://1drv.ms/u/s!Aq335pidOrBV4II3db1xzFESWcuLvw

Let me know if you spot anything not working. Also any comments are welcomed.

**Comment by danoe** [comment]
Works great so far! :) Thanks so much.

**Comment by danoe** [comment]
Could the new queue system also be used for these one off downloads/uploads? Currently i usually start a bunch of them and they all run at the same time, so the bwlimit doesn't really help. If they would run one after the other that would solve this.

**Comment by kapitainsky** [comment]
Good to hear that so far so good.

The way how queue is designed is that it needs tasks so what you ask can't  work out of the box - well it is software so all could be done but sometimes it is just too much work.
But I can imagine adding button "save and run: (ALT-something ?):). Then we would need task name - e.g. temp_date_time_unique_id. It should not be a big deal to delete all tasks later - tasks list supports multiselect. Or temp_* tasks can be auto deleted on quit unless user changes their name. There are many ways how to skin a cat.

At the moment I am still working on this new release. I rather don't add it before it is out but definitely we can think about solving it later as I can see that it can be useful for many people.

**Comment by kapitainsky** [comment]
When new version is released - in week or two time - please create new issue. Not now - it would confuse people as stable release does not have queue.

**Comment by danoe** [comment]
Absolutely :) Thanks for all your hard work.
My idea for a shortcut would be ALT+s, i'd say ALT+q but on MacOS Option+q might be too close to CMD+q :)

Auto clean and/or button to clean temporary tasks sounds like a good idea, like you said with some kind of prefix and generated names.

**Comment by csss1234** [comment]
Would this allow pausing and resuming jobs? I have the slowest internet in the world, and trying to upload just ~10gigs of essential files takes 2+ days. I need to be able to pause the uploading occasionally to reclaim that bandwidth. I know I could just limit the bandwidth, and I do, but a pause and resume feature would be great.

**Comment by kapitainsky** [comment]
yes. when you run transfer task you can stop it at any time and then start again. It is not pause sensu stricto but rather stop/start rclone. Queue is persistent between applications restarts and only fully finished tasks are removed from it. So you can even turn computer off and then start all again.

**Comment by kapitainsky** [comment]
if you are on windows try latest beta - link in the first post of #93

**Comment by dia3olik** [comment]
This is handy but with many small files it can leads to time wasted since it has to do many checks again on restart, right?

**Comment by kapitainsky** [comment]
and how not to "waste" this time?

**Comment by dia3olik** [comment]
he he maybe pausing the underlying rclone process?

**Comment by kapitainsky** [comment]
I doubt it is something that can be implemented across all operating systems. And don't see any real practical benefit - very few niche cases would benefit vs a lot of coding and testing. If you have to stop and start the same job again you have to live with small overhead needed to recheck already transferred files. It can be further reduced by using `--fast-list` on some remotes.

**Comment by csss1234** [comment]
Thanks for quick reply. Im on a mac

**Comment by kapitainsky** [comment]
Here you are the latest beta for macOS:

https://1drv.ms/u/s!Aq335pidOrBV4IJJ-XhPhrvqr9KEBw

In case you described save your transfers as tasks and add them to the queue. Queue will process them one by one and can be stopped and re-started at any time.

**Comment by kapitainsky** [comment]
if you notice something not working or have any comments would be great if you let me know.

**Comment by kapitainsky** [comment]
@csss1234 if you have period of time you don't want your tasks to consume bandwidth you could also use scheduler and `--max-duration` option,

as per documentation:

```
--max-duration=TIME
Rclone will stop scheduling new transfers when it has run for the duration specified.
Defaults to off.
When the limit is reached any existing transfers will complete.
Rclone won’t exit with an error if the transfer limit is reached.
```

so let's say you only want to run your task between 1am and 8am - you schedule it to run daily at 1am and add `--max-duration 7h` in extra options.
